Fix Typing Errors in Docs, Comments and other Explanations.

faze-geek opened the pull request:
Ran the `codespell` library over the `lpython` codebase and found various typos. I've gone through all of them to avoid any false positives.









P.S. - Having setup `lpython` a couple of days back on windows system, I tried to not make code related changes and get the builds successfully working. Hope all tests pass here!

certik commented on tests/dictionary1.py:
```diff
@@ -1,7 +1,7 @@
     x: dict[i32, i32]
     x = {1: 2, 3: 4}
-    # x = {1: "2", "3": 4} -> sematic error
+    # x = {1: "2", "3": 4} -> semantic error
```
This change changes the hash of this file, which then causes to fail the `./run_tests.py`. To fix that, just do:
```
./run_test.py -u
```
To regenerate the tests and commit the changes.

faze-geek replied:
I have updated the reference files using `python run_tests.py -u` which modified existing files and created new files as well. Didn't expect the updated files to be these many in number. Could you kindly check if this is as expected !?

czgdp1807 replied:
Done. Whenever you face this kind of situation in LPython follow the steps below (assuming you are under `lpython`'s root),

1. `cd tests/reference`.
2. `rm ./*` (execute this command very carefully, because it deletes many files).
3. `cd ../../`
4. `python run_tests.py -u`
5. `git add .`
6. `git commit -m "Updated reference tests"`.

faze-geek replied:
Thanks for the assist. I will keep this in mind.

faze-geek replied:
> 2\. rm ./*

I want to run this, but I guess the command is for a linux terminal. I'll have to figure out it's windows equivalent.

czgdp1807 reviewed (changes requested):
This is good but update the reference tests by running `python run_tests.py -u` as Ondrej suggested in his review comment. Thanks.

certik commented:
Can you please rebase or merge the latest master and resolve conflicts?

faze-geek commented:
> Can you please rebase or merge the latest master and resolve conflicts?

Conflicts are tricky!
As in should the "infile_hash" be updated through this pr (`"4532f5384a81ab702e30f6b957032bbd5665773d8cbaefb47c55de2d"`) or stay as it is in main (`"51c244e84b614bbe3d536fd561b1016eb904fb766aeb67e521e9fe65"`)

czgdp1807 commented:
Again, nothing much is to be done. Rebase your branch on top of `main` of LPython. Then do `git add .`, `git rebase --continue` (because the conflicts in generated reference files, not much to think about). Do a clean build and follow the steps in https://github.com/lcompilers/lpython/pull/1430#discussion_r1072285841. And push. That's all.

faze-geek commented:
> Again, nothing much is to be done. Rebase your branch on top of `main` of LPython. 

```
Auto-merging src/libasr/ASR.asdl
Auto-merging src/libasr/codegen/asr_to_llvm.cpp
Auto-merging src/lpython/semantics/python_ast_to_asr.cpp
Auto-merging src/lpython/semantics/python_comptime_eval.h
CONFLICT (content): Merge conflict in src/lpython/semantics/python_comptime_eval.h
Auto-merging src/runtime/impure/lfortran_intrinsics.c
CONFLICT (content): Merge conflict in src/runtime/impure/lfortran_intrinsics.c
Auto-merging src/runtime/impure/lfortran_intrinsics.h
error: could not apply 6a09783b0... Add `RealToCharacter` casting to `str()`
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 6a09783b0... Add `RealToCharacter` casting to `str()`

C:\Users\kunni\lpython>git status
interactive rebase in progress; onto bd1288378
Last command done (1 command done):
   pick 6a09783b0 Add `RealToCharacter` casting to `str()`
Next commands to do (1271 remaining commands):
   pick 4f32d5cc8 Add `IntegerToCharacter` casting to `str()`
   pick e87522278 Add `LogicalToCharacter` casting to `str()`
  (use "git rebase --edit-todo" to view and edit)
You are currently rebasing branch 'Fix_Typos' on 'bd1288378'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   src/libasr/ASR.asdl
        modified:   src/libasr/codegen/asr_to_llvm.cpp
        modified:   src/lpython/semantics/python_ast_to_asr.cpp
        modified:   src/runtime/impure/lfortran_intrinsics.h
        modified:   src/runtime/lpython_builtin.py

Unmerged paths:
  (use "git restore --staged <file>..." to unstage)
  (use "git add <file>..." to mark resolution)
        both modified:   src/lpython/semantics/python_comptime_eval.h
        both modified:   src/runtime/impure/lfortran_intrinsics.c
```

I have fetched origin and run `git rebase origin/main`. It resolves existing conflicts but leads to more conflicts (and added files) which is strange.
Should I manually remove the conflicts on github itself?

czgdp1807 commented:
I would recommend you do `git rebase --abort` and then `git merge main`. We will squash and merge this PR here on Github, so don't worry about rebasing things. Thanks.

faze-geek commented:
> I would recommend you do `git rebase --abort` and then `git merge main`.

`git merge main` gives `Already up to date`. I think the conflicts could be manually removed on github then you can squash and merge.

certik commented:
Thanks @faze-geek for the patch and thanks @czgdp1807 for finishing it!
